libs/com.etendorx.utils.auth/src/main/java/com/etendorx/utils/auth/key/context/FilterContext.java

```diff
@@ -26,27 +26,48 @@
 import javax.servlet.http.HttpServletResponse;
 import java.io.IOException;
 import java.util.Map;
+import java.util.Optional;
 
 @Component
 public class FilterContext extends OncePerRequestFilter {
 
+  private static final String AUTH = "AUTH";
+
   @Override
   protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
     String token = request.getHeader("X-TOKEN");
-    if (token != null && !token.isBlank()) {
-      setUserContextFromToken(token);
+    Optional<String> service = Optional.ofNullable(request.getHeader("X-SERVICE"));
+    Optional<String> username = Optional.ofNullable(request.getHeader("X-USERNAME"));
```

Comment:
For what use case do you get the username and service from a header? They should only be in the payload of the authentication request. In this case these data are included in the token.